### PR TITLE
commands/new; Allow Ansible Operator to be created outside GOPATH

### PR DESCRIPTION
**Pull request by dymurray**

### CODE REVIEW

File: commands/operator-sdk/cmd/new.go [hunk 1]
```diff
@@ -121,20 +121,25 @@ func mustBeNewProject() {
 }
 
 // repoPath checks if this project's repository path is rooted under $GOPATH and returns project's repository path.
-func repoPath() string {
-	gp := os.Getenv(gopath)
-	if len(gp) == 0 {
-		cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("$GOPATH env not set"))
-	}
+// repoPath field on generator is used primarily in generation of Go operator. For Ansible we will set it to cwd
+func repoPath(operatorType string) string {
```
Comment (shawn-hurley) [inline commands/operator-sdk/cmd/new.go]: I do not believe you need to pass this into the function, you should be able to get it as it is a package var

Comment (fanminshi) [reply commands/operator-sdk/cmd/new.go]: maybe we should have a enum for operatorType instead of string?

File: commands/operator-sdk/cmd/new.go [hunk 2]
```diff
@@ -121,20 +121,25 @@ func mustBeNewProject() {
...
 	wd := mustGetwd()
-	// check if this project's repository path is rooted under $GOPATH
-	if !strings.HasPrefix(wd, gp) {
-		cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("project's repository path (%v) is not rooted under GOPATH (%v)", wd, gp))
-	}
-	// compute the repo path by stripping "$GOPATH/src/" from the path of the current directory.
-	rp := filepath.Join(string(wd[len(filepath.Join(gp, src)):]), projectName)
-	// strip any "/" prefix from the repo path.
-	return strings.TrimPrefix(rp, string(filepath.Separator))
+	if isGoOperator(operatorType) {
```
Comment (joelanford) [inline commands/operator-sdk/cmd/new.go]: Can this just be: 
```
if operatorType == "go" { ... }
```
Or, if we use an enum: 
```
if operatorType == OperatorTypeGo { ... }
```

**Comment by hasbro17** [comment]
LGTM

**Pull request merged**